File: 53. Maximum Subarray/main.py
class Solution:

    def maxSubArray(self, A):
        if not A:
            return 0

        curSum = maxSum = A[0]
        for num in A[1:]:
            curSum = max(num, curSum + num)
            maxSum = max(maxSum, curSum)

        return maxSum

File: 53. Maximum Subarray/main.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_so_far = nums[0], curr_max = nums[0], start = 0, end = 0;
        for(int i=1; i<nums.size(); i++){
            curr_max = max(curr_max+nums[i], nums[i]);
            max_so_far = max(max_so_far, curr_max);
        }
        return max_so_far;
    }
};
